keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -115,26 +115,24 @@ class ArtifactListener(
           launch {
             val lastRecordedVersion = getLatestStoredVersion(artifact)
             log.debug("Last recorded version of $artifact: $lastRecordedVersion")
+
             val artifactSupplier = artifactSuppliers.supporting(artifact.type)
-            val latestArtifact = artifactSupplier.getLatestArtifact(artifact.deliveryConfig, artifact)
-            val latestVersion = latestArtifact?.let { artifactSupplier.getFullVersionString(it) }
-            log.debug("Latest available version of $artifact: $latestVersion")
+            val latestVersion = artifactSupplier.getLatestArtifact(artifact.deliveryConfig, artifact)
+            log.debug("Latest available version of $artifact: ${latestVersion?.version}")
 
             if (latestVersion != null) {
               val hasNew = when {
                 lastRecordedVersion == null -> true
-                latestVersion != lastRecordedVersion -> {
-                  artifact.versioningStrategy.comparator.compare(lastRecordedVersion, latestVersion) > 0
+                latestVersion.version != lastRecordedVersion -> {
+                  artifact.versioningStrategy.comparator.compare(lastRecordedVersion, latestVersion.version) > 0
                 }
                 else -> false
               }
 
               if (hasNew) {
-                log.debug("$artifact has a missing version $latestVersion, persisting.")
-                val status = artifactSupplier.getReleaseStatus(latestArtifact)
-                repository.storeArtifact(artifact.name, artifact.type, latestVersion, status)
-                // send an event when the artifact is stored, so we can fetch its metadata
-                publisher.publishEvent(ArtifactSaved(latestArtifact, status))
+                log.debug("$artifact has a missing version ${latestVersion.version}, persisting.")
+                val enrichedArtifact = artifactSupplier.addMetadata(latestVersion.normalized())
+                repository.storeArtifactVersion(enrichedArtifact)
```

Comment:
Instead of doing one write to `ARTIFACT_VERSIONS` to store the basics here, and another as a reaction to the `ArtifactSaved` event just below, consolidate in a single call to `storeArtifactVersion()`.